Python/specialize.c

```diff
@@ -696,8 +708,38 @@ _Py_Specialize_LoadAttr(PyObject *owner, _Py_CODEUNIT *instr, PyObject *name)
             goto fail;
         }
         case PROPERTY:
-            SPECIALIZATION_FAIL(LOAD_ATTR, SPEC_FAIL_ATTR_PROPERTY);
-            goto fail;
+        {
+            _PyLoadMethodCache *cache = (_PyLoadMethodCache *)(instr + 1);
+            assert(Py_TYPE(descr) == &PyProperty_Type);
+            PyObject *fget = _PyProperty_PropGet(descr);
+            if (fget == NULL || Py_TYPE(fget) != &PyFunction_Type) {
+                SPECIALIZATION_FAIL(LOAD_ATTR, SPEC_FAIL_EXPECTED_ERROR);
+                goto fail;
+            }
+            PyFunctionObject *func = (PyFunctionObject *)fget;
+            PyCodeObject *fcode = (PyCodeObject *)func->func_code;
+            int kind = function_kind(fcode);
+            if (kind != SIMPLE_FUNCTION) {
+                SPECIALIZATION_FAIL(LOAD_ATTR, kind);
+                goto fail;
+            }
+            if (fcode->co_argcount != 1) {
+                SPECIALIZATION_FAIL(LOAD_ATTR, SPEC_FAIL_WRONG_NUMBER_ARGUMENTS);
+                goto fail;
+            }
+            int version = _PyFunction_GetVersionForCurrentState(func);
+            if (version == 0 || version != (uint16_t)version) {
+                SPECIALIZATION_FAIL(LOAD_ATTR, SPEC_FAIL_OUT_OF_VERSIONS);
```

Comment:
Technically, if `version != (uint16_t)version` we are out of bounds, not out of versions.
`SPECIALIZATION_FAIL(LOAD_ATTR, version == 0 ? SPEC_FAIL_OUT_OF_VERSIONS : SPEC_FAIL_OUT_OF_RANGE`